### PR TITLE
Swagger cleanup for credential endpoint

### LINKED ISSUE #1091: Review Swagger Docs: Credentials section
## Specify type:
- Enhancement

### Priority:
- Medium
___

## Description:
Review the credentials APIs for correct definition and descriptions. Improve the associated example information
___

## Acceptance Criteria:

- [ ] Verify that swagger docs are awesome for the Credentials API

### PR DESCRIPTION
**Pull request by kholdaway**
- Add authorization header to all API (except /token)

Closes #1091

### CODE REVIEW

File: docs/swagger.yml [hunk 1]
```diff
@@ -69,33 +80,35 @@ paths:
...
-      description: "List all credential for use when connecting to a target system."
+      description: "List all credential."
       operationId: "listCredentials"
       produces:
       - "application/json"
       parameters:
         - name: "name"
           in: "query"
-          description: "A list of names representing credentials"
+          description: "Filter list by a list of credential names."
```
Comment (abaiken) [inline docs/swagger.yml]: This sounds a little weird.

File: docs/swagger.yml [hunk 2]
```diff
@@ -145,13 +160,15 @@ paths:
...
+        401:
+          description: "Not authorized"
         404:
           description: "Credential not found"
     put:
       tags:
         - "Credential"
-      summary: "Update an existing credential"
-      description: "Update a credential for use when connecting to a target system."
+      summary: "Full update an existing credential"
```
Comment (abaiken) [inline docs/swagger.yml]: Should this say 'Fully update an existing credential'?

File: docs/swagger.yml [hunk 3]
```diff
@@ -177,13 +194,49 @@ paths:
...
           description: "Invalid input"
+        401:
+          description: "Not authorized"
+        404:
+          description: "Credential not found"
+    patch:
+      tags:
+        - "Credential"
+      summary: "Partial update of an existing credential"
+      description: "Partial update an existing credential."
```
Comment (abaiken) [inline docs/swagger.yml]: Same ? as above. 'Partially update' or 'Partial update of an' instead of 'Partial update an'.

File: docs/swagger.yml [hunk 4]
```diff
@@ -177,13 +194,49 @@ paths:
...
+      parameters:
+      - name: "cred_id"
+        in: "path"
+        description: "ID of credential to update"
+        required: true
+        type: "integer"
+        format: "int64"
+      - in: "body"
+        name: "body"
+        description: "Portion of credential object that should overlay existing credential object."
```
Comment (abaiken) [inline docs/swagger.yml]: I feel like the descriptions of updates should be somewhat cohesive: 
ie. full update body description says: 'Credential object to replace existing credential object'
so maybe this should say: 'Credential id to replace existing credential id' or something similar?

File: docs/swagger.yml [hunk 5]
```diff
@@ -916,6 +971,7 @@ paths:
       summary: "Obtain a user token."
       description: "Obtain a user token."
       operationId: "obtainToken"
+      security: []
```
Comment (chambridge) [inline docs/swagger.yml]: Status endpoint also doesn't require a token

File: docs/swagger.yml (hunk 4 above)
Comment (kholdaway) [reply docs/swagger.yml]: Its not a replacement of credential id.

File: docs/swagger.yml (hunk 1 above)
Comment (kholdaway) [reply docs/swagger.yml]: Fixed.

File: docs/swagger.yml (hunk 2 above)
Comment (kholdaway) [reply docs/swagger.yml]: Yep

File: docs/swagger.yml (hunk 3 above)
Comment (kholdaway) [reply docs/swagger.yml]: Fixed

File: docs/swagger.yml (hunk 5 above)
Comment (kholdaway) [reply docs/swagger.yml]: Fixed

Comment (chambridge) [reply docs/swagger.yml]: looks like you updated log out instead of status.

Comment (kholdaway) [reply docs/swagger.yml]: Fixed.

**Pull request merged**